src/features/ProductPage/components/Product.jsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import {useNavigate} from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+const Product = ({productObj}) => {
+  const navigate = useNavigate();
+
+
+  const onProductClicked = (event, id) => {
+    navigate(`/products/${id}`);
+  };
+
+  return (
+    <div className="flex flex-col gap-2">
+      <div className="w-40 h-40 shadow flex flex-row items-center justify-center bg-slate-300">
+        <img alt="product-image" src={productObj.image}
+          className="max-w-full max-h-full w-auto h-auto" />
+      </div>
+      <button onClick={(e) => onProductClicked(
```

Comment:
move to a defined, callbacked function, rather than an anonymous one.

Reply:
done